src/js/view/controls/tizen/tizen-controlbar.ts

```diff
@@ -0,0 +1,251 @@
+import Controlbar from 'view/controls/controlbar';
+import { cloneIcons } from 'view/controls/icons';
+import TimeSlider from 'view/controls/components/timeslider';
+import CustomButton from 'view/controls/components/custom-button';
+import button, { Button } from 'view/controls/components/button';
+import { SimpleTooltip } from 'view/controls/components/simple-tooltip';
+import { toggleClass } from 'utils/dom';
+import type { PlayerAPI } from 'types/generic.type';
+import type ViewModel from 'view/view-model';
+import type NextUpToolTip from 'view/controls/nextuptooltip';
+
+type ControlbarElement = HTMLElement | Button | TimeSlider | any;
+
+function div(classes: string): HTMLElement {
+    const element = document.createElement('div');
+    element.className = `jw-reset ${classes}`;
+    return element;
+}
+
+const appendChildren = (container: HTMLElement, elements: ControlbarElement[]) => {
```

Comment:
Are we checking for e.element() because this function takes two types of arguments (JS views and HTML elements)? Rather than allowing it to pass the JS view or the element, we should set this in a way so it's always an element being passed in (single responsibility principle).

Reply:
👍 This gave me an idea to extract the extract out the elements in a new function to pass to appendChildren.